### PR TITLE
[BACKLOG-36843]-Accessibility-Buttons and Keycodes

### PR DESCRIPTION
**Pull request by phalguni-a**
@pentaho/wcag , please review

### CODE REVIEW

File: impl/client/src/main/javascript/web/util/_a11y.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    end: 35,
+    home: 36,
+    arrowLeft: 37,
+    arrowUp: 38,
+    arrowRight: 39,
+    arrowDown: 40,
+    delete: 46
+  };
+
+  var _isToggleButton = false;
```
Comment (dcleao) [inline impl/client/src/main/javascript/web/util/_a11y.js:38]: This module variable is not the best way to solve the issue. There are simpler, safer, cleaner solutions.

For example, create an internal function which receives a parameter that indicates the type of button:
```js
function makeAccessibleButton(elem, isToggleButton) {
   // ...
}

return {
    keyCodes: keyCodes,

    makeAccessibleActionButton: function(elem) { 
       makeAccessibleButton(elem, false);
    },

    makeAccessibleToggleButton: function(elem) { 
       makeAccessibleButton(elem, true);
    }
  };
```

**Review by dcleao — CHANGES_REQUESTED** [review]

File: impl/client/src/main/javascript/web/util/_a11y.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  function makeAccessibleActionButton(elem) {
+    elem.role = "button";
+    elem.tabIndex = 0;
+
+    elem.addEventListener('keydown', actionButtonKeydownHandler);
+    elem.addEventListener('keyup', actionButtonKeyupHandler);
+
+    function actionButtonKeydownHandler(event) {
```
Comment (dcleao) [inline impl/client/src/main/javascript/web/util/_a11y.js:47]: Note suggested character case of "KeyDown": `actionButtonKeyDownHandler`.

File: impl/client/src/main/javascript/web/util/_a11y.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        event.preventDefault();
+      } else if (event.keyCode === keyCodes.enter) {
+        // If enter is pressed, activate the button
+        event.preventDefault();
+        handleToggleButton();
+        elem.click();
+      }
+    }
+
+    function actionButtonKeyupHandler(event) {
```
Comment (dcleao) [inline impl/client/src/main/javascript/web/util/_a11y.js:61]: Note suggested character case of "KeyUp": `actionButtonKeyUpHandler`.

**Comment by phalguni-a** [comment]
@dcleao , Addressed the required. Can you please review now.

**Review by hv-vamsi — APPROVED** [review]

**Review by ddiroma — APPROVED** [review]

**Pull request merged**